Penteract/Assets/Scripts/AIDuke.cpp

```diff
@@ -429,133 +520,112 @@ void AIDuke::OnAnimationEvent(StateMachineEnum stateMachineEnum, const char* eve
 			dukeCharacter.state = DukeState::STUNNED;
 		}
 	}
 
 	if (!dukeCharacter.isAlive) {
-		/*std::string curState = "Run";
-		if (animation->GetCurrentState()->name == "Idle") curState = "Idle";
-		DeleteAttackCollider();
-		deathType = (rand() % 2 == 0) ? true : false;
-		if (state == AIState::ATTACK) {
-			if (deathType) {
-				animation->SendTrigger(curState + "Death1");
-				animation->SendTriggerSecondary("AttackDeath1");
-			}
-			else {
-				animation->SendTrigger(curState + "Death2");
-				animation->SendTriggerSecondary("AttackDeath2");
-			}
-		}
-		else if (state == AIState::IDLE) {
-			if (deathType) {
-				animation->SendTrigger("IdleDeath1");
-			}
-			else {
-				animation->SendTrigger("IdleDeath1");
-			}
-		}
-		else if (state == AIState::RUN) {
-			if (deathType) {
-				animation->SendTrigger("RunDeath1");
-			}
-			else {
-				animation->SendTrigger("RunDeath2");
-			}
-		}
-		else if (state == AIState::STUNNED) {
-			if (deathType) {
-				animation->SendTrigger("StunnedDeath1");
-			}
-			else {
-				animation->SendTrigger("StunnedDeath2");
-			}
-		}
-		else if (state == AIState::PUSHED) {
-			if (deathType) {
-				animation->SendTrigger("HurtDeath1");
-			}
-			else {
-				animation->SendTrigger("HurtDeath2");
-			}
-		}
+		movementScript->Stop();
+		dukeCharacter.StopShooting();
+		dukeCharacter.compAnimation->SendTrigger(dukeCharacter.compAnimation->GetCurrentState()->name + dukeCharacter.animationStates[Duke::DUKE_ANIMATION_STATES::DEATH]);
 
-		if (audios[static_cast<int>(AudioType::DEATH)]) audios[static_cast<int>(AudioType::DEATH)]->Play();
+		// TODO: play audio and VFX
+		//if (audios[static_cast<int>(AudioType::DEATH)]) audios[static_cast<int>(AudioType::DEATH)]->Play();
 		ComponentCapsuleCollider* collider = GetOwner().GetComponent<ComponentCapsuleCollider>();
 		if (collider) collider->Disable();
 
-		agent->RemoveAgentFromCrowd();
-		if (gruntCharacter.beingPushed) gruntCharacter.beingPushed = false;*/
+		dukeCharacter.agent->RemoveAgentFromCrowd();
+		if (dukeCharacter.beingPushed) dukeCharacter.beingPushed = false;
 		dukeCharacter.state = DukeState::DEATH;
 	}
 }
 
 void AIDuke::EnableBlastPushBack() {
-	if (dukeCharacter.state != DukeState::INVULNERABLE) {
+	if (dukeCharacter.state != DukeState::INVULNERABLE && dukeCharacter.state != DukeState::CHARGE) {
 		dukeCharacter.beingPushed = true;
 		dukeCharacter.state = DukeState::PUSHED;
-		//if (animation->GetCurrentState()) animation->SendTrigger(animation->GetCurrentState()->name + "Hurt");
+		dukeCharacter.compAnimation->SendTrigger(dukeCharacter.compAnimation->GetCurrentState()->name + dukeCharacter.animationStates[Duke::DUKE_ANIMATION_STATES::PUSHED]);
+		dukeCharacter.StopShooting();
 		CalculatePushBackRealDistance();
+
+		OnShieldInterrupted();
+
 		// Damage
 		if (playerController->playerOnimaru.level2Upgrade) {
 			dukeCharacter.GetHit(playerController->playerOnimaru.blastDamage + playerController->GetOverPowerMode());
```

Comment:
hitTaken = true needs to be add after GetHit

Reply:
but there's no hitTaken here?